includes/transformer/class-base.php

```diff
@@ -30,9 +31,37 @@ abstract class Base {
 	 *
 	 * @param WP_Post|WP_Comment $wp_object The WordPress object
 	 *
-	 * @return Base_Object
+	 * @return Base
 	 */
-	public static function transform( $object ) {
+	public static function transform( $object ) { // phpcs:ignore Universal.NamingConventions.NoReservedKeywordParameterNames.objectFound
```

Comment:
Do we need this filterable too? You can filter it already on the Factory level!

https://github.com/Automattic/wordpress-activitypub/blob/d94d064acbc42d371065f3d3839cd949eee0b7c7/includes/transformer/class-factory.php#L50

Reply:
The collection and outbox endpoints don't use the Factory, as far as I can tell. That's why I added it in the `Base` class as well.

But: I'd actually suggest to always use the Factory (and not `Post::transform()`, which also returns a transformer).

Before, you'd do `Post::transform( $post )->to_object()`, but if you're going to check that `Post::transform()` really _does_ return a transformer, then you can't "chain" anyways. (And even then, you could've also done `Factory::get_transformer( $post )->to_object()`. But I'm guessing this code "grew organically" and that's how it ended up.)

So maybe I should just replace `$transformer = Post_Transformer::transform( $post );` with `$transformer = Factory::get_transformer( $post );`. Then the code would be the same always and we can drop the filter from the `Base` class (and always use the Factory instead).

Reply:
I see, I agree with you on always using the transformer! Maybe force it, by protecting the `__construct`.

Reply:
Maybe we can keep the chain and have the sanity check in the `to_*` methods?